packages/block-editor/src/components/rich-text/index.js

```diff
@@ -289,6 +306,23 @@ export function RichTextWrapper(
 		anchorRef.current?.focus();
 	}
 
+	let shouldDisableEditing = false;
+	if ( blockBindings ) {
+		const blockTypeAttributes = getBlockType( blockName ).attributes;
+		for ( const [ attribute, args ] of Object.entries( blockBindings ) ) {
+			// If any of the attributes with source "rich-text" is part of the bindings,
+			// has a source with `lockAttributesEditing`, disable it.
+			if (
+				blockTypeAttributes?.[ attribute ]?.source === 'rich-text' &&
+				getBlockBindingsSource( args.source.name )
+					?.lockAttributesEditing
+			) {
+				shouldDisableEditing = true;
```

Comment:
Could this potentially all be calculated in the useSelect callback? I'm not sure how much the bindings change, which would cause the component to re-render.

Reply:
I can give that a try. In this initial version, bindings shouldn't change unless users modify them through the Code Editor. But I'll try to move it in the `useSelect` callback.

Reply:
I've done it in [this commit](https://github.com/WordPress/gutenberg/pull/58085/commits/aaf8652d0e1facf6f1cd7431a4f96819e1f59c1b). Let me know if that's not what you had in mind.